crates/cast/src/lib.rs

```diff
@@ -307,22 +318,40 @@ pub async fn handle_wait_for_tx<T>(
 
 pub fn print_formatted(
     mut output: Vec<(&str, String)>,
-    int_format: bool,
+    value_format: ValueFormat,
     json: bool,
     error: bool,
 ) -> Result<()> {
-    if !int_format {
-        output = output
-            .into_iter()
-            .map(|(key, value)| {
+    output = output
+        .into_iter()
+        .map(|(key, value)| match value_format {
+            ValueFormat::Default => {
                 if let Ok(int_value) = U256::from_dec_str(&value) {
                     (key, format!("{int_value:#x}"))
                 } else {
                     (key, value)
                 }
-            })
-            .collect();
-    }
+            }
+            ValueFormat::Int => {
+                if let Ok(int_value) = U256::from_dec_str(&value) {
+                    (key, format!("{int_value}"))
+                } else if let Ok(field) = FieldElement::from_str(value.as_str()) {
+                    (key, format!("{field}"))
+                } else {
+                    (key, value)
+                }
+            }
+            ValueFormat::Hex => {
+                if let Ok(int_value) = U256::from_dec_str(&value) {
+                    (key, format!("{int_value:#x}"))
+                } else if let Ok(field) = FieldElement::from_str(value.as_str()) {
+                    (key, format!("{field:#x}"))
+                } else {
+                    (key, value)
+                }
+            }
+        })
+        .collect();
```

Comment:
does this logic make sense @THenry14? `Default` is when neither `int-format` or `hex-format` is specified.

Reply:
and i assume that when neither is specified we receive values in appropriate format from source

Reply:
The logic seems okay. If we had a trait implementing .format(), this maybe could be simplified though